frontend/src/components/templates/Form/HideSectionsContainer.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {
+  Children,
+  createContext,
+  FunctionComponent,
+  useEffect,
+  useState,
+} from 'react';
+import Button from '../../atoms/hyperTexts/Button';
+import Loader from '../../atoms/Loader';
+
+export type HideSectionsContextType = {
+  setReadyForNextSteps: (v: boolean) => void;
+};
+
+export const HideSectionsContext = createContext<HideSectionsContextType>({
+  setReadyForNextSteps: () => null,
+});
+
+type Props = {
+  SubmissionPanel: FunctionComponent;
+};
+
+export const HideSectionsContainer: FunctionComponent<Props> = ({
+  SubmissionPanel,
+  children,
+}) => {
+  const [showOnlyFirstStep, setShowOnlyFirstStep] = useState(false);
+  const [isShowNextStepButtonDisabled, setIsShowNextStepButtonDisabled] =
+    useState(false);
+  const [areNextStepsLoading, setAreNextStepLoading] = useState(false);
+  const [readyForNextSteps, setReadyForNextSteps] = useState(true);
+
+  useEffect(() => {
+    if (showOnlyFirstStep) {
+      setIsShowNextStepButtonDisabled(true);
+    }
+  }, [setIsShowNextStepButtonDisabled, showOnlyFirstStep]);
+
+  const showNextStep = () => {
+    setIsShowNextStepButtonDisabled(true);
+    setAreNextStepLoading(true);
+    setTimeout(() => {
+      setAreNextStepLoading(false);
+      setShowOnlyFirstStep(false);
+    }, 1800);
+  };
+
+  useEffect(() => {
```

Comment:
Same as above, this useEffect should be removed.

You have 2 handlers for setting `showOnlyFirstStep` and `readyForNextSteps` that have side effect. It should be explicit in the code.

Do not use the `useState` setters if there are side effect on other states or logic to do. Instead use your own handlers.

Here is an example implementation that should replace the call for the useState setters in this component and jsx: 
```ts

  const handleOnlyShowFirstStep = (show: boolean) => {
    if (show) {
      setIsOnlyFirstStepShown(true);
      setIsShowNextStepButtonDisabled(!isReadyForNextSteps);
      return;
    }

    setIsOnlyFirstStepShown(!isReadyForNextSteps);
  };
  ```
```ts
  const handleReadyForNextStep = (ready: boolean) => {
    if (isOnlyFirstStepShown) {
      setIsShowNextStepButtonDisabled(!ready);
      return;
    }

    onlyShowFirstStep(!ready);
  };
  ```